src/routers/AppRouter.tsx

```diff
@@ -0,0 +1,31 @@
+import { HashRouter, Routes, Route } from 'react-router-dom';
+import App from '../App';
+import Header from '../components/Header/Header';
+import AddStock from '../pages/AddStock';
+import Login from '../pages/Login';
+import ProtectedRoute from 'pages/ProtectedRoute';
+import Signup from '../pages/Signup';
+import WrongRoutePage from '../pages/WrongRoutePage';
+
+const AppRouter = () => {
+  return (
+    <HashRouter>
+      <Header />
+      <Routes>
+        <Route path="/" element={<App />} />
+        <Route path="/login" element={<Login />} />
+        <Route path="/signup" element={<Signup />} />
+        <Route path="*" element={<WrongRoutePage />} />
+        <Route
+          path="/addStock"
+          element={
+            <ProtectedRoute user={'authorizedUser'}>
```

Comment:
Two possible solutions here could be to (1) just make `user` a boolean variable that says whether or not the user is logged in, or (2) make `user` an object with a boolean `loggedIn` field. We could add a `useEffect` hook above the return statement that makes a `GET` request to the backend `/api/v1/authenticated` url, and use that to set some `loggedIn` or `user` piece of state.

Reply:
The user can be `null` at first, could check if it's an object after fetching.